File: leetcode-js/334-increasing-triplet-subsequence.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var increasingTriplet = function(nums) {
    let c1 = Infinity;
    let c2 = Infinity;

    for (const n of nums) {
        if (n <= c1) {
            c1 = n;
        } else if (n <= c2) {
            c2 = n;
        } else {
            return true;
        }
    }

    return false;
};

File: leetcode-java/334-increasing-triplet-subsequence.java
public class Solution {
    public boolean increasingTriplet(int[] nums) {
        int smallest = Integer.MAX_VALUE;
        int smaller = Integer.MAX_VALUE;
        for (int n : nums) {
            if (n <= smallest) {
                smallest = n;
            } else if (n <= smaller) {
                smaller = n;
            } else {
                return true;
            }
        }
        return false;
    }
}
